Server/JokesOnYou.Web.Api/Controllers/TagController.cs

```diff
@@ -1,19 +1,65 @@
-﻿using JokesOnYou.Web.Api.Services.Interfaces;
+﻿using JokesOnYou.Web.Api.DTOs;
+using JokesOnYou.Web.Api.Services.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
 using System;
-using System.Collections.Generic;
-using System.Linq;
+using System.Security.Claims;
 using System.Threading.Tasks;
 
 namespace JokesOnYou.Web.Api.Controllers
 {
+    [Route("api/[controller]/[action]")]
+    [ApiController]
+    [Authorize]
     public class TagController : ControllerBase
     {
         private readonly ITagService _tagService;
+        private readonly ILogger _logger;
+        private readonly UserManager<ApplicationIdentity> _userManager;
 
-        public TagController(ITagService tagService)
+        public TagController(ITagService tagService, ILogger<TagController> logger, UserManager<ApplicationIdentity> userManager)
         {
             _tagService = tagService;
+            _logger = logger;
+            _userManager = userManager;
+        }
+
+        [HttpDelete]
+        public async Task<IActionResult> DeleteTag(int tagId)
+        {
+            var tag = await _tagService.GetTag(tagId);
+            string userId = _userManager.GetUserId(User);
+            if (tag.UserReplyDTO.Id == userId || User.IsInRole("Admin"))
```

Comment:
This is a short circuit opperand, if first condition is true we are not evaluating if the used role is "Admin" is this intended?

Reply:
This same comment applies to the other implementation

Reply:
Yes this is intended.
Because if either the User is the owner or the User is an admin s/he can delete the tag (for now).